### PR TITLE
refactor: remove OpError, use ErrBox everywhere 

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

File: core/errors.rs [hunk 1]
```diff
@@ -3,72 +3,110 @@
...
 impl<T> AnyError for T where T: Any + Error + Send + Sync + Sized + 'static {}
 
 #[derive(Debug)]
-pub struct ErrBox(Box<dyn AnyError>);
+pub enum ErrBox {
+  Simple {
+    class: &'static str,
+    message: Cow<'static, str>,
+  },
+  Boxed(Box<dyn AnyError>),
```
Comment (ry) [inline core/errors.rs:24]: no class in the boxed case?

Comment (piscisaureus) [reply core/errors.rs:24]: The class is determined when it is needed. See `cli/errors.js`.

Comment (piscisaureus) [reply core/errors.rs:24]: In particular the function `rust_err_to_json()`

File: core/core_isolate.rs [hunk 2]
```diff
@@ -123,6 +134,7 @@ pub struct CoreIsolateState {
   pub(crate) js_macrotask_cb: Option<v8::Global<v8::Function>>,
   pub(crate) pending_promise_exceptions: HashMap<i32, v8::Global<v8::Value>>,
   pub(crate) js_error_create_fn: Box<JSErrorCreateFn>,
+  pub rust_err_to_json_fn: &'static dyn RustErrToJsonFn,
```
Comment (ry) [inline core/core_isolate.rs:137]: This would be unnecessary if every ErrBox contained a "class" string.

Comment (piscisaureus) [reply core/core_isolate.rs:137]: Well *only* if you specify the error class at every single location where an error is converted to an ErrBox (and there are a lot!). You'd have to opt out of using `.into()` and the ? operator entirely.

**Pull request merged**